src/h5web/visualizations/shared/utils.ts

```diff
@@ -45,37 +45,52 @@ export function computeVisSize(
     : { width, height: width / aspectRatio };
 }
 
+export function getExtent(
+  values: number[],
+  initialExtent?: [number, number, number]
+): [number, number, number] {
+  return values.reduce((acc, val) => {
+    const [accMin, accMax, accPositiveMin] = acc;
+    return [
+      Math.min(val, accMin),
+      Math.max(val, accMax),
+      val > 0 ? Math.min(val, accPositiveMin) : accPositiveMin,
+    ];
+  }, initialExtent || [values[0], values[0], Infinity]);
+}
+
 export function getDomain(
   values: number[],
-  scaleType: ScaleType = ScaleType.Linear
+  scaleType: ScaleType = ScaleType.Linear,
+  errors?: number[]
 ): Domain | undefined {
   if (values.length === 0) {
     return undefined;
   }
 
-  const [min, max, positiveMin] = values.reduce<
-    [number, number, number | undefined]
-  >(
-    (acc, val) => {
-      const [accMin, accMax, accPositiveMin] = acc;
-      return [
-        accMin > val ? val : accMin,
-        accMax < val ? val : accMax,
-        val > 0 && (accPositiveMin === undefined || accPositiveMin > val)
-          ? val
-          : accPositiveMin,
-      ];
-    },
-    [values[0], values[0], undefined]
-  );
+  if (errors && errors.length !== values.length) {
+    throw new Error(
+      'Errors when computing domain should have the same size as values'
+    );
+  }
+
+  const [min, max, positiveMin] = errors
+    ? values.reduce(
+        (acc, val, i) => {
+          const err = errors[i];
+          return getExtent([val - err, val, val + err], acc);
+        },
+        [values[0], values[0], Infinity]
+      )
+    : getExtent(values);
```

Comment:
I am not extremely happy about how this turned out. Happy to get some ideas.

Reply:
Hmm, took me a minute to understand, but I think it's quite elegant. Another solution would be for `getExtent` to only be the inner function of the `reduce`, and to call it once for the value, and two more times for the value with errors:

```
values.reduce((acc, val, i) => {
  const res = getExtent(acc, value);
  return !errors ? res : getExtent(getExtent(res, value - err), value + err);
});
```

Not as elegant, but perhaps simpler.